### PR TITLE
Merge hdf4-examples repo into library

### PR DESCRIPTION
**Pull request by byrnHDF**
Just like hdf5 process allow examples to be built with library and provide examples as a standalone product ditributed with the binaries.

### CODE REVIEW

**Review by hyoklee — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

File: HDF4Examples/CMakePresets.json [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  ],
+  "configurePresets": [
+    {
+      "name": "ci-base-examples",
+      "hidden": true,
+      "cacheVariables": {
+        "CPACK_PACKAGE_VERSION": "3.10.2",
+        "HDF4_NAMESPACE": {"type": "STRING", "value": "hdf5::"},
+        "HDF4_PACKAGE_NAME": {"type": "STRING", "value": "hdf5"},
+        "H5EX_BUILD_TESTING": "ON"
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply HDF4Examples/CMakePresets.json:14]: Fixed

File: HDF4Examples/config/cmake/HDFMacros.cmake [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      set (CMAKE_TEST_OUTPUT_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${HDF_CFG_NAME})
+      set (CMAKE_PDB_OUTPUT_DIRECTORY
+          ${PROJECT_BINARY_DIR}/bin CACHE PATH "Single Directory for all pdb files."
+      )
+    else ()
+      set (CMAKE_TEST_OUTPUT_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+    endif ()
+  else ()
+    # if we are externally configured, but the project uses old cmake scripts
+    # this may not be set and utilities like H5detect will fail
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply HDF4Examples/config/cmake/HDFMacros.cmake:31]: maybe I think I saw it once.

[the remark this answers is not in the document]
Comment (byrnHDF) [reply HDF4Examples/config/cmake/HDFMacros.cmake:31]: changed statement

File: HDF4Examples/config/cmake/HDFMacros.cmake [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        OUTPUT  "${dest}"
+        COMMAND "${CMAKE_COMMAND}"
+        ARGS     -E copy_if_different "${src}" "${dest}"
+        DEPENDS "${src}"
+    )
+    list (APPEND ${target}_list "${dest}")
+endmacro ()
+
+macro (HDF_DIR_PATHS package_prefix)
+  option (H5EX_USE_GNU_DIRS "TRUE to use GNU Coding Standard install directory variables, FALSE to use historical settings" FALSE)
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply HDF4Examples/config/cmake/HDFMacros.cmake:83]: fixed

File: HDF4Examples/config/cmake/HDFMacros.cmake [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  if (H5EX_USE_GNU_DIRS)
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply HDF4Examples/config/cmake/HDFMacros.cmake:84]: fixed

File: HDF4Examples/CMakePresets.json [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+  "include": [
+    "config/cmake-presets/hidden-presets.json"
+  ],
+  "configurePresets": [
+    {
+      "name": "ci-base-examples",
+      "hidden": true,
+      "cacheVariables": {
+        "CPACK_PACKAGE_VERSION": "3.10.2",
+        "HDF4_NAMESPACE": {"type": "STRING", "value": "hdf5::"},
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply HDF4Examples/CMakePresets.json]: fixed

File: HDF4Examples/CMakePresets.json [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        "HDF4_PACKAGE_NAME": {"type": "STRING", "value": "hdf5"},
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply HDF4Examples/CMakePresets.json]: fixed

**Pull request merged**